adyenofficial.php

```diff
@@ -366,31 +375,45 @@ public function createAdyenOrderStatuses()
      */
     public function createWaitingForPaymentOrderStatus()
     {
-        if (!Configuration::get('ADYEN_OS_WAITING_FOR_PAYMENT')) {
-            $order_state = new OrderState();
-            $order_state->name = array();
+        $orderStateConfigurationId = Configuration::get('ADYEN_OS_WAITING_FOR_PAYMENT');
+
+        $orderState = false;
+        if ($orderStateConfigurationId) {
+            $orderState = $this->orderStateAdapter->getOrderStateById($orderStateConfigurationId);
+        }
+
+        // In case order state does not exist in the database anymore
+        if (empty($orderState)) {
```

Comment:
I think this should also work? Just simplifying a bit more.
```suggestion
        if ($orderStateConfigurationId) {
            $orderState = $this->orderStateAdapter->getOrderStateById($orderStateConfigurationId);
        }

        // In case order state does not exist in the database anymore
        if (!empty($orderState)) {
```

Reply:
We only want to create a new order state if the $orderState is empty.

The only thing that could be removed is the `$orderState = false;` declaration at the beginning, but I don't like to check undefined variables even though in php it would work.